visma/functions/constant.py

```diff
@@ -48,14 +48,31 @@ def __radd__(self, other):
         return self + other
 
     def __add__(self, other):
-        if self.isZero():   # if one of them is Empty, we can return the other one even it is Empty too because we need at least one to be returned.
+        if isinstance(other, Constant):
+            if other.power == self.power:
+                if self.before == '-':
```

Comment:
What is `before`?
Can you give an example for above?

Reply:
`before` is an object of _Function_ Class, it indicates what symbol precedes a particular token. Like in `1 + 2 + 3`, `1` gets tokenized as _Constant_ (a derived-class of _Function_, say `alpha` ). So, `alpha.before` will indicate be " " .Likewise beta.before will be `+`. (if `beta` is token corresponding to 2). Generally, These values become specified after passing tokens to `getLevelVariables` and are used to determine which token is to added/subtracted to which one.